app/jobs/maintenance_tasks/task_job.rb

```diff
@@ -90,7 +90,7 @@ def after_perform
     end
 
     def on_error(error)
-      @ticker.persist
+      @ticker.persist if defined?(@ticker)
```

Comment:
Why the variable would not be defined? Because `before_perform` did run?

Reply:
`before_perform` runs but has an exception before `@ticker` is set, on line 57:

https://github.com/Shopify/maintenance_tasks/blob/42fd4ee3cec2286f4a1f53f31f51c6402101e587/app/jobs/maintenance_tasks/task_job.rb#L55-L62

Reply:
At first I just used `@ticker&.persist` but it didn't pass the test because we use verbose mode. We could also add a `@ticker = nil` at the beginning of `before_perform` too but I felt that would feel weird.